### ISSUE TITLE
Nested configuration options are not shown

### ISSUE THREAD

**Issue by laeubi** [eclipse/lemminx-maven#349]
Tycho has a nested configuration option `<target>`
https://tycho.eclipseprojects.io/doc/latest/target-platform-configuration/target-platform-configuration-mojo.html#target

but this is not shown as a completion:
![grafik](https://user-images.githubusercontent.com/1331477/212837118-c3008a62-2837-4c8e-bc27-fab930a3be7a.png)

Also if one starts typing nothing is shown:
![grafik](https://user-images.githubusercontent.com/1331477/212837430-fb690a77-8006-4a22-bd45-51807386ce5c.png)

Also inside the tag, no completion is provided (should be `<artifact>`, `<file>` or `<url>`):
![grafik](https://user-images.githubusercontent.com/1331477/212837730-6cd9f7a0-806e-40ce-b232-23ddbec0f021.png)

**Comment by mickaelistria** [comment]
The fact that `target` isn't shown is indeed a bug. I'm looking at it.
The fact that we don't see nested elements is because the type of the parameter in the Mojo is not a bean or a list but a Xpp3Dom. And as a result, there is nothing that can be inferred from the mojo.
I tried a while ago to make the `target` parameter in TargetPlatformConfigurationMojo better typed, but it's tricky because it doesn't obey usual Maven patterns: the name of the children matters, and children can be of multiple types according to element name. I gave up and hoped that the complement/hover documentation would usually be helping enough.
